ART-2445 Add find-bugs mode to check for blocker bugs

thegreyd opened the pull request:
## Links
- https://issues.redhat.com/browse/ART-2445
- Previous PR https://github.com/openshift/aos-cd-jobs/pull/2519

## Summary:
This PR adds a new mode to find-bugs 'blocker', which looks up blocker bugs for a group. 
- By default cve trackers are included
- Also a new flag `--exclude-status` which lets you exclude bug status

## TODO:
- [X] The list of component filters in story is different from what we have in bugzilla.yml, ~~make sure that bugzilla.yml is updated with that~~ After discussion with @sosiouxme we agreed to use the "default" component filter list in bugzilla.yml without any modification, since that is what ART uses in other places. The list could be changed/modified in the future accordingly.
- [x] prepare-release and promote job PR https://github.com/openshift/aos-cd-jobs/pull/2534

## Test
- `./elliott -g openshift-4.7 find-bugs --mode blocker --report`
- `./elliott -g openshift-4.7 find-bugs --mode blocker --exclude-status ON_QA --report`

thegreyd commented on elliottlib/cli/find_bugs_cli.py:
```diff
@@ -21,20 +21,27 @@
               help="Bugzilla IDs to add, required for LIST mode.")
 @click.option("--cve-trackers",
               required=False,
+              default=None,
```
I wasn't sure if there was a way to set this to False if the default was True. Like `--cve-trackers=False`, if not the case I'll revert this change. Part of the exploration of "different modes have different defaults, but if a user explicitly sets it then use that".

sosiouxme replied:
i'm pretty sure `--cve-trackers=False` works fine

thegreyd commented on elliottlib/cli/find_bugs_cli.py:
```diff
@@ -21,20 +21,27 @@
               type=click.Choice(constants.VALID_BUG_STATES),
-              help="Status of the bugs")
+              help="Include bugs of this status")
+@click.option("--exclude-status", 'exclude_status',
```
This was added so we can use blocker in prepare-release job with `--exclude-status ON_QA`

sosiouxme commented on elliottlib/cli/find_bugs_cli.py:
```diff
@@ -123,49 +137,74 @@ def find_bugs_cli(runtime, advisory, default_advisory_type, mode, check_builds,
     bz_data = runtime.gitdata.load_data(key='bugzilla').data
     bzapi = bzutil.get_bzapi(bz_data)
 
+    # Some bugs should goes to CPaaS so we should ignore them
```
this comment seems out of place. and wherever it might go, I think I'd prefer something along the lines of "filter out bugs ART does not manage"

sosiouxme commented on elliottlib/cli/find_bugs_cli.py:
```diff
@@ -123,49 +137,74 @@ def find_bugs_cli(runtime, advisory, default_advisory_type, mode, check_builds,
-            status = ['MODIFIED']
+    if mode in ['sweep', 'qe', 'blocker']:
+        if not cve_trackers:
+            if mode == 'blocker':
+                cve_trackers = True
+            else:
+                cve_trackers = False
```
Seems a bit verbose, how about:
```suggestion
            cve_trackers = mode == 'blocker'
```

thegreyd replied:
Not sure about that, I prefer the first one since the branching makes it very clear that cve_trackers value varies by mode. The value could be bool, or string doesn't matter, the structure would be: `if not param: if mode == 'X': param = Y`

thegreyd replied:
Also this is soon to be refactored into individual function for each mode, so we would get rid of this anyway :)

sosiouxme commented on elliottlib/cli/find_bugs_cli.py:
```diff
@@ -123,49 +137,74 @@ def find_bugs_cli(runtime, advisory, default_advisory_type, mode, check_builds,
+            else:
+                cve_trackers = False
+
+        if not status:
```
how about a comment?
```suggestion
        if not status:  # use default status filter according to mode
```

sosiouxme reviewed (commented):
alright, i have nits, but works well and LGTM aside from the nits (which I'll leave to your discretion)

thegreyd commented on elliottlib/bzutil.py:
```diff
@@ -217,18 +217,19 @@ def create_placeholder(bz_data, kind):
     :param bz_data: The Bugzilla data dump we got from our bugzilla.yaml file
     :param status: The status(es) of bugs to search for
     :param search_filter: Which search filter from bz_data to use if multiple are specified
+    :param flag: A specific flag to filter out bugs
```
Added
```python
:param flag: The Bugzilla flag (string) of bugs to search for. Flags are similar to status but are categorically
    different. https://bugzilla.readthedocs.io/en/latest/using/understanding.html#flags
```

thegreyd replied:
But I'm not sure what else we can rename the param, since it's called flag everywhere in bz.

thegreyd commented:
> alright, i have nits, but works well and LGTM aside from the nits (which I'll leave to your discretion)

Thank you for the review! I've addressed the comments.
